actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -542,7 +542,7 @@ def current_page?(options, check_parameters: false)
 
         return false unless request.get? || request.head?
 
-        check_parameters ||= !options.is_a?(String) && options.try(:delete, :check_parameters)
+        check_parameters ||= options.respond_to?(:to_hash) && options.try(:delete, :check_parameters)
```

Comment:
I think that is fine. To `check_parameter` be inside a Parameters instance either the end user would have to submit that information in the request (which I think it is a really bad idea) or the developer would have to merge the Paremeters with the option (in this case they should be using the named argument).